tensorboard/webapp/webapp_data_source/tb_feature_flag_data_source_types.ts

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Injectable} from '@angular/core';
+
+export type FeatureValue = string | boolean | number | string[];
+
+@Injectable()
+export abstract class TBFeatureFlagDataSource {
+  abstract getFeatures(): {
+    enabledExperimentalPlugins: string[];
+    [featureKey: string]: FeatureValue;
```

Comment:
This seems very generic. Related to the comment above, how is this used? Is it currently being used by anything? If not, how will it be used in the future? Is there a concrete plan? Give some examples.